modules/access_level/variables.tf

```diff
@@ -16,68 +16,75 @@
+  type        = string
 }
 
 variable "ip_subnetworks" {
   description = "Condition - A list of CIDR block IP subnetwork specification. May be IPv4 or IPv6. Note that for a CIDR IP address block, the specified IP address portion must be properly truncated (i.e. all the host bits must be zero) or the input is considered malformed. For example, \"192.0.2.0/24\" is accepted but \"192.0.2.1/24\" is not. Similarly, for IPv6, \"2001:db8::/32\" is accepted whereas \"2001:db8::1/32\" is not. The originating IP of a request must be in one of the listed subnets in order for this Condition to be true. If empty, all IP addresses are allowed."
-  type        = "list"
-  default     = []
+  type        = list(string)
+  default     = null
 }
 
 variable "required_access_levels" {
   description = "Condition - A list of other access levels defined in the same Policy, referenced by resource name. Referencing an AccessLevel which does not exist is an error. All access levels listed must be granted for the Condition to be true."
-  type        = "list"
-  default     = []
+  type        = list(string)
+  default     = null
 }
 
 variable "members" {
   description = "Condition - An allowed list of members (users, service accounts). The signed-in identity originating the request must be a part of one of the provided members. If not specified, a request may come from any user (logged in/not logged in, etc.). Formats: user:{emailid}, serviceAccount:{emailid}"
-  type        = "list"
-  default     = []
+  type        = list(string)
+  default     = null
 }
 
 variable "negate" {
   description = "Whether to negate the Condition. If true, the Condition becomes a NAND over its non-empty fields, each field must be false for the Condition overall to be satisfied."
-  default     = "false"
+  type        = bool
+  default     = false
 }
 
 variable "require_screen_lock" {
   description = "Condition - Whether or not screenlock is required for the DevicePolicy to be true."
-  default     = "false"
+  type        = bool
+  default     = false
 }
 
 variable "allowed_encryption_statuses" {
   description = "Condition - A list of allowed encryptions statuses. An empty list allows all statuses."
-  type        = "list"
-  default     = []
+  type        = list(string)
+  default     = null
 }
 
 variable "allowed_device_management_levels" {
   description = "Condition - A list of allowed device management levels. An empty list allows all management levels."
-  type        = "list"
-  default     = []
+  type        = list(string)
+  default     = null
 }
 
 variable "minimum_version" {
   description = "The minimum allowed OS version. If not set, any version of this OS satisfies the constraint. Format: \"major.minor.patch\" such as \"10.5.301\", \"9.2.1\"."
-  default     = ""
+  type        = string
+  default     = null
 }
 
 variable "os_type" {
   description = "The operating system type of the device."
-  default     = ""
+  type        = string
+  default     = null
```

Comment:
Same

Reply:
Thank you, Ihor!
[Resolved](https://github.com/terraform-google-modules/terraform-google-vpc-service-controls/commit/63c5a8a98dbafd8436547a6d994293cafb99c825)